webapp/packages/core-session-localization/src/SessionLocalizationService.ts

```diff
@@ -8,27 +8,30 @@
 import { Dependency, injectable } from '@cloudbeaver/core-di';
 import { LocalizationService } from '@cloudbeaver/core-localization';
 import { SessionResource } from '@cloudbeaver/core-root';
+import { SettingsLocalizationService } from '@cloudbeaver/core-settings-localization';
 
 @injectable()
 export class SessionLocalizationService extends Dependency {
-  constructor(private readonly sessionResource: SessionResource, private readonly localizationService: LocalizationService) {
+  constructor(
+    private readonly sessionResource: SessionResource,
+    private readonly localizationService: LocalizationService,
+    private readonly settingsLocalizationService: SettingsLocalizationService,
+  ) {
     super();
 
-    this.sessionResource.beforeLoad.addHandler(() => {
-      this.sessionResource.setDefaultLocale(this.localizationService.currentLanguage);
-    });
     this.sessionResource.onDataUpdate.addHandler(this.syncLanguage.bind(this));
-    this.localizationService.onChange.addHandler(key => {
-      this.sessionResource.setDefaultLocale(key);
-      this.sessionResource.changeLanguage(key);
+    this.settingsLocalizationService.settingsProvider.onChange.addHandler(data => {
+      if (data.key === 'language') {
+        this.sessionResource.changeLanguage(data.value).catch(() => {});
```

Comment:
please add error message here so we get an error in console once something went wrong with

`await this.graphQLService.sdk.changeSessionLanguage({ locale });`